pkg/internal/controller/controller_test.go

```diff
@@ -169,6 +172,108 @@ var _ = Describe("controller", func() {
 			close(done)
 		}, 10.0)
 
+		It("should not error when source.channel is passed", func(done Done) {
+			eventCount := 0
+			// channel to be closed when event is processed
+			processed := make(chan struct{})
+			// source channel to be injected
+			ch := make(chan event.GenericEvent, 1)
+
+			ctx, cancel := context.WithCancel(context.TODO())
+			defer cancel()
+
+			// event to be sent to the channel
+			p := &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
+			}
+			evt := event.GenericEvent{
+				Object: p,
+			}
+
+			ins := &source.Channel{Source: ch}
+			ins.DestBufferSize = 1
+			Expect(inject.StopChannelInto(ctx.Done(), ins)).To(BeTrue())
+
+			// send the event to the channel
+			ch <- evt
+
+			ctrl.startWatches = []watchDescription{{
+				src: ins,
+				handler: handler.Funcs{
+					GenericFunc: func(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+						defer GinkgoRecover()
+						eventCount++
+
+						if eventCount > 0 {
```

Comment:
I don't think we need the `eventCount` variable in any way, success is indicated by channel closing which is currently just hidden behind a complicate way of expressing `if true`